internal/olm/operator/registry/configmap/registry_test.go

```diff
@@ -0,0 +1,458 @@
+						Objects: []*unstructured.Unstructured{
+							{
+								Object: map[string]interface{}{"val1": "val1"},
+							},
+							{
+								Object: map[string]interface{}{"val2": "va2"},
+							},
+						},
+						CSV: &v1alpha1.ClusterServiceVersion{
+							Spec: v1alpha1.ClusterServiceVersionSpec{
+								Version: version.OperatorVersion{
+									Version: semver.SpecVersion,
+								},
+							},
+						},
+					},
+				},
+				Client: &client.Client{
+					KubeClient: fakeclient,
+				},
+			}
+		})
+
+		It("should return true if a registry is stale if there are no configmaps in the cluster", func() {
+			fakeclient := fake.NewFakeClient()
+			rr.Client.KubeClient = fakeclient
+			temp, err := rr.IsRegistryDataStale(context.TODO(), testns)
+			Expect(temp).Should(BeTrue())
+			Expect(err).Should(BeNil())
+		})
+
+		It("should return true if a registry is stale", func() {
+			temp, err := rr.IsRegistryDataStale(context.TODO(), testns)
+			Expect(temp).Should(BeTrue())
+			Expect(err).Should(BeNil())
+		})
+
+		It("should return true if a registry is stale", func() {
+			fakeclient := fake.NewFakeClient(
+				&corev1.ConfigMapList{
+					Items: []corev1.ConfigMap{
+						{
+							ObjectMeta: metav1.ObjectMeta{
+								Name:      getRegistryConfigMapName(pkgName) + "-package",
+								Namespace: testns,
+								Labels:    makeRegistryLabels(pkgName),
+							},
+						},
+						{
+							ObjectMeta: metav1.ObjectMeta{
+								Name:      "config2",
+								Namespace: testns,
+								Labels:    makeRegistryLabels(pkgName),
+							},
+						},
+					},
+				},
+				newRegistryDeployment(pkgName, testns),
+				newRegistryService(pkgName, testns),
+			)
+			rr.Client.KubeClient = fakeclient
+			temp, err := rr.IsRegistryDataStale(context.TODO(), testns)
+			Expect(temp).Should(BeTrue())
+			Expect(err).Should(BeNil())
+		})
+
+		It("should return true if the binary data does not have a filekey", func() {
+			binarydata, err := makeObjectBinaryData(struct {
+				val1 string
+				val2 string
+			}{
+				val1: "val1",
+				val2: "val2",
+			}, "userInput")
+			fakeclient := fake.NewFakeClient(
+				&corev1.ConfigMapList{
+					Items: []corev1.ConfigMap{
+						{
+							BinaryData: binarydata,
+							ObjectMeta: metav1.ObjectMeta{
+								Name:      getRegistryConfigMapName(pkgName) + "-package",
+								Namespace: testns,
+								Labels:    makeRegistryLabels(pkgName),
+							},
+						},
+						{
+							ObjectMeta: metav1.ObjectMeta{
+								Name:      "config2",
+								Namespace: testns,
+								Labels:    makeRegistryLabels(pkgName),
+							},
+						},
+					},
+				},
+				newRegistryDeployment(pkgName, testns),
+				newRegistryService(pkgName, testns),
+			)
+			rr.Client.KubeClient = fakeclient
+			temp, err := rr.IsRegistryDataStale(context.TODO(), testns)
+			Expect(temp).Should(BeTrue())
+			Expect(err).Should(BeNil())
+		})
+
+		It("should return true if the binary data does not have a filekey", func() {
```

Comment:
Is this the same test? make sure the `It` descriptions describe what we're testing.